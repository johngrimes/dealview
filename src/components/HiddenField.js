import React from 'react'
import PropTypes from 'prop-types'

class HiddenField extends React.Component {
  render() {
    const value = this.props.value ? this.props.value : ''

    return <input id={this.props.name} type='hidden' value={value} />
  }
}

HiddenField.propTypes = {
  name: PropTypes.string.isRequired,
  value: PropTypes.string,
}

export default HiddenField
