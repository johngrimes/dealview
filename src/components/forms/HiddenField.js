// @flow

import React from 'react'

type Props = {
  +name: string,
  +value?: string
}

class HiddenField extends React.Component {
  props: Props

  render() {
    const value = this.props.value ? this.props.value : ''

    return (
      <input id={this.props.name} type='hidden' value={value} />
    )
  }
}

export default HiddenField
