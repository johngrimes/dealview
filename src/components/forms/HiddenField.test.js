/* global expect */

import HiddenField from './HiddenField.js'

import React from 'react'
import { shallow } from 'enzyme'

describe('HiddenField', () => {
  it('should render', () => {
    const props = {
      name: 'someInput',
      value: 'something'
    }
    const wrapper = shallow(<HiddenField {...props} />)
    expect(wrapper).toMatchSnapshot()
  })
})
