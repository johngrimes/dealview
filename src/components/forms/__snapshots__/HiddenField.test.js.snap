// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HiddenField should render 1`] = `
<input
  id="someInput"
  type="hidden"
  value="something"
/>
`;
